File: 226-invert-binary-tree/226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
const invertTree = function(root) {
    if (!root) return root;
    const q = [root];
    
    while (q.length) {
        const node = q.shift();
        const temp = node.left;
        node.left = node.right;
        node.right = temp;
        if (node.left) q.push(node.left);
        if (node.right) q.push(node.right);
    }
    
    return root;
};

File: 226-invert-binary-tree/226-invert-binary-tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} root
# @return {TreeNode}
def invert_tree(root)
    if root
        temp = root.left
        root.left = root.right
        root.right = temp
        invert_tree root.left
        invert_tree root.right
        root
    end
end
